src/clr-angular/forms/datalist/datalist-container.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ContentChild, Inject } from '@angular/core';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { DynamicWrapper } from '../../../clr-angular/utils/host-wrapping/dynamic-wrapper';
+import { ControlIdService } from '../common/providers/control-id.service';
+import { FocusService } from '../common/providers/focus.service';
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { NgControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { ClrLabel } from '../common/label';
+
+@Component({
+  selector: 'clr-datalist-container',
+  template: `
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [ngClass]="controlClass()">
+      <div class="clr-input-wrapper">
+        <div class="clr-input-group" [class.clr-focus]="focus">
```

Comment:
Just a question on my part here. Is there any benefit to using `[class.clr-focus]` over an `[ngClass]` object here?

I don't know which one is better or if it even matters in this case.

Reply:
`[class.clr-focus]` is a shorthand so you don't have to create a function as well, yield the same result but this is more declarative in the template itself.

This PR is superceded by #4121 and will be closed by it.